aws-ssm-document/src/main/java/com/amazonaws/ssm/document/UpdateHandler.java

```diff
@@ -68,6 +73,14 @@
         safeLogger.safeLogDocumentInformation(model, callbackContext, request.getAwsAccountId(), request.getSystemTags(), logger);
 
         // Only Tags are handled in Update Handler. Other properties of the Document resource are CreateOnly.
+        // Verify no CreateOnly properties are modified
+        if(isCreateOnlyModified(model, previousModel)) {
+            return ProgressEvent.failed(
+                    model,
+                    context,
+                    HandlerErrorCode.NotUpdatable, "Create-Only Property cannot be updated");
```

Comment:
is that a good message?

Reply:
I think this message is an ok generic message
Another possibility would be to have a message more specific to the property being modified, but that would require checking each property individually

Reply:
that's true but may not be the best customer experience. we can modify that later